### DISCUSSION TITLE
Hooray for 0.53.2

### DISCUSSION THREAD

**Discussion by mattneub** [navidrome/navidrome#3317] [General]
The big change in 0.53.2 that has affected me personally is the fix for randomness when asking the server for some random albums. I use this feature every day, and I had a nagging feeling all this time that I wasn't getting as much randomness as I was entitled to; more or less the same albums kept showing up in the list, even though I have 1000 albums. I didn't want to say anything about it because, proverbially, an appearance of non-randomness is itself a possible outcome of randomness, so I wasn't sure of my ground. But now — wow! The difference is huge. Very nice catch, @deluan!

**Comment by deluan** [comment]
Thanks! Actually the old implementation was too random, it would give a completely random page every single time, even when you asked for a different page, which could include items from previous calls/pages. That's because it was not keeping the seed between calls/pages.

This has been a known issue for a long time (at least since 2021: https://github.com/navidrome/navidrome/issues/749), and AFAIK most (all?) Subsonic API implementations that uses SQLite have the same issue. 

The ingenious solution by @gqgs on #1841 was the turning point to work on a proper implementation for true, seeded randomness.

Anyway, glad you liked it :)
